mrp_production_serial_matrix/wizards/mrp_production_serial_matrix.py

```diff
@@ -0,0 +1,342 @@
+                rec.production_id,
+                rec.finished_lot_ids,
+            )
+            rec.line_ids = False
+            rec.write({"line_ids": [(0, 0, x) for x in matrix_lines]})
+
+    def button_validate(self):
+        self.ensure_one()
+        if self.lot_selection_warning_count > 0:
+            raise UserError(
+                _("Some issues has been detected in your selection: %s")
+                % self.lot_selection_warning_msg
+            )
+        mos = self.env["mrp.production"]
+        current_mo = self.production_id
+        for fp_lot in self.finished_lot_ids:
+            # Apply selected lots in matrix and set the qty producing
+            current_mo.lot_producing_id = fp_lot
+            current_mo.qty_producing = 1.0
+            current_mo._set_qty_producing()
+            for move in current_mo.move_raw_ids:
+                if move.product_id.tracking in ["serial", "lot"]:
+                    # We filter using the lot nane because the ORM sometimes
+                    # is not storing correctly the finished_lot_id in the lines
+                    # after passing through the `_onchange_finished_lot_ids`
+                    # method.
+                    matrix_lines = self.line_ids.filtered(
+                        lambda l: (
+                            l.finished_lot_id == fp_lot
+                            or l.finished_lot_name == fp_lot.name
+                        )
+                        and l.component_id == move.product_id
+                    )
+                    if matrix_lines:
+                        self._amend_reservations(move, matrix_lines)
+                        self._consume_selected_lots(move, matrix_lines)
+
+            # Complete MO and create backorder if needed.
+            mos += current_mo
+            res = current_mo.button_mark_done()
+            backorder_wizard = self.env["mrp.production.backorder"]
+            if isinstance(res, dict) and res.get("res_model") == backorder_wizard._name:
+                # create backorders...
+                lines = res.get("context", {}).get(
+                    "default_mrp_production_backorder_line_ids"
+                )
+                wizard = backorder_wizard.create(
+                    {
+                        "mrp_production_ids": current_mo.ids,
+                        "mrp_production_backorder_line_ids": lines,
+                    }
+                )
+                wizard.action_backorder()
+
+                backorder_ids = (
+                    current_mo.procurement_group_id.mrp_production_ids.filtered(
+                        lambda mo: mo.state not in ["done", "cancel"]
+                    )
+                )
+                current_mo = backorder_ids[0] if backorder_ids else False
+                if not current_mo:
+                    break
+            else:
+                break
+
+        # TODO: not specified lots: auto create lots?
```

Comment:
Just as a suggestion, here we could check if mos is empty to prevent being redirected to a blank MO list, for example when no final lot is specified. Then the user could simply be redirected to self.production_id.